### ISSUE TITLE
java.lang.Throwable is shown instead of name of concrete exception class implementation  

### ISSUE THREAD

**Issue by DmytroBatyuk** [akaita/RxJava2Debug#6]
Original stacktrace:

>  FATAL EXCEPTION: pool-13-thread-2
                                                                 Process: com.myapp.dev, PID: 7141
                                                                 io.reactivex.exceptions.OnErrorNotImplementedException: TEST exception!!!
                                                                     at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
                                                                     at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
                                                                     at io.reactivex.internal.subscribers.LambdaSubscriber.onError(LambdaSubscriber.java:79)
                                                                     at io.reactivex.internal.subscribers.BasicFuseableSubscriber.onError(BasicFuseableSubscriber.java:101)
                                                                     at io.reactivex.internal.subscribers.BasicFuseableSubscriber.fail(BasicFuseableSubscriber.java:111)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:66)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:69)
                                                                     at io.reactivex.internal.operators.flowable.FlowableFilter$FilterSubscriber.tryOnNext(FlowableFilter.java:74)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapConditionalSubscriber.tryOnNext(FlowableMap.java:129)
                                                                     at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnConditionalSubscriber.runAsync(FlowableObserveOn.java:637)
                                                                     at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
                                                                     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                                     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                                     at java.lang.Thread.run(Thread.java:761)
                                                                  Caused by: java.lang.UnsupportedOperationException: TEST exception!!!
                                                                     at com.myapp.dev.home.HomeViewModel$observeProfilePicture$1.apply(HomeViewModel.kt:42)
                                                                     at com.myapp.dev.home.HomeViewModel$observeProfilePicture$1.apply(HomeViewModel.kt:13)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:64)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:69) 
                                                                     at io.reactivex.internal.operators.flowable.FlowableFilter$FilterSubscriber.tryOnNext(FlowableFilter.java:74) 
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapConditionalSubscriber.tryOnNext(FlowableMap.java:129) 
                                                                     at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnConditionalSubscriber.runAsync(FlowableObserveOn.java:637) 
                                                                     at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176) 
                                                                     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
                                                                     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
                                                                     at java.lang.Thread.run(Thread.java:761) `

Generated by library stacktrace:

> FATAL EXCEPTION: pool-15-thread-2
                                                                 Process: com.myapp.dev, PID: 7637
                                                                 java.lang.Throwable: TEST exception!!!
                                                                     at com.myapp.dev.ui.home.HomeViewModel$observeProfilePicture$1.apply(HomeViewModel.kt:42)
                                                                     at com.myapp.dev.ui.home.HomeViewModel$observeProfilePicture$1.apply(HomeViewModel.kt:13)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:64)
                                                                     at hu.akarnokd.rxjava2.debug.FlowableOnAssembly$OnAssemblySubscriber.onNext(FlowableOnAssembly.java:61)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:69)
                                                                     at hu.akarnokd.rxjava2.debug.FlowableOnAssembly$OnAssemblySubscriber.onNext(FlowableOnAssembly.java:61)
                                                                     at io.reactivex.internal.operators.flowable.FlowableFilter$FilterSubscriber.tryOnNext(FlowableFilter.java:74)
                                                                     at hu.akarnokd.rxjava2.debug.FlowableOnAssembly$OnAssemblyConditionalSubscriber.tryOnNext(FlowableOnAssembly.java:103)
                                                                     at io.reactivex.internal.operators.flowable.FlowableMap$MapConditionalSubscriber.tryOnNext(FlowableMap.java:129)
                                                                     at hu.akarnokd.rxjava2.debug.FlowableOnAssembly$OnAssemblyConditionalSubscriber.tryOnNext(FlowableOnAssembly.java:103)
                                                                     at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnConditionalSubscriber.runAsync(FlowableObserveOn.java:637)
                                                                     at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
                                                                     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
                                                                     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
                                                                     at java.lang.Thread.run(Thread.java:761)
                                                                  Caused by: java.lang.Throwable
                                                                  Caused by: java.lang.Throwable

Android version: 7.1.1
RxJava version: 2.1.7

**Comment by akaita** [comment]
Yeah, that's true.  
I looked into it in the past and I don't think I can completely solve it. 

I'm thinking, I could just put the exception class and message as the message of the synthetic exception. That'd give us all the info we need, right?

**Comment by DmytroBatyuk** [comment]
Sure. It would be much better

**Comment by vincentbrison** [comment]
Hi,
Would it be possible to put the exception class and message as the message of the synthetic exception? It would greatly improve the understandability of stack traces from crashlytics.

For now a npe would simply result in a java.lang.Throwable.

Thank you

**Comment by akaita** [comment]
@vincentbrison yeah, that's exactly what `I'm thinking, I could just put the exception class and message as the message of the synthetic exception` means xD 
I plan on giving a new push this week to get this and some other stuff published 👍

**Comment by vincentbrison** [comment]
Thanks for the quick feedback, that is good to know :). Since it was discussed around February, I was wondering if it would still be the case.

**Comment by akaita** [comment]
Solved in [1.4.0](https://github.com/akaita/RxJava2Debug/releases/tag/1.4.0).

It will now say something like:
```
Caused by: java.lang.Throwable: caused by java.lang.NullPointerException: The mapper function returned a null value.
	at com.akaita.java.rxjava2debug.exampleandroid.MainActivity.generateUnhandledException(MainActivity.kt:95)
	at com.akaita.java.rxjava2debug.exampleandroid.MainActivity.access$generateUnhandledException(MainActivity.kt:31)
	at com.akaita.java.rxjava2debug.exampleandroid.MainActivity$onCreate$3.onClick(MainActivity.kt:41)
	at android.view.View.performClick(View.java:6597)
	at android.view.View.performClickInternal(View.java:6574)
	at android.view.View.access$3100(View.java:778)
	at android.view.View$PerformClick.run(View.java:25885)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6669)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

Thanks for reporting.

**Comment by vincentbrison** [comment]
Thanks a lot!
